### DISCUSSION TITLE
OpenGL? Vulkan? Other GPU Interfaces?

### DISCUSSION THREAD

**Discussion by d3x0r** [wasmerio/wasmer#2841] [General]
Vulkan?  Custom system abstract APIs?

What if... I have this system abstraction library .... https://github.com/d3x0r/sack  

it's sort of like a syscall interface similar to what wasmer runtime does for other system services (sockets,files, ...) .

Emscripten SDL is actually meant to target a HTML Canvas (no?) And implementing a full canvas API is surely not a great solution?

The driver for 3D things in SACK to WebGL/Empscriten canvas was actually very light (very little code); because mostly it's just linking to opengl calls - but then - does wasmer support that?

What does a system interface module look like?

(Wow there's a lot of red highlighted words in the above :) )

**Comment by SamuraiCrow** [comment]
The WASI standard doesn't presently support those interfaces.
